### PR TITLE
Remove SpecialMatrices & callbacks saving fix

### PR DESCRIPTION
**Pull request by MSeeker1340**
The SpecialMatrices package is no longer available due to the recent package compatibility capping and it is preventing tests from executing. The only affected test script is multivariate_geometric.jl which for some reason is not included in the test suite. I have added it and also made some minor fixes to the test script (note that the last test seems to be broken).

### CODE REVIEW

**Comment by MSeeker1340** [comment]
Since the fix for saveat handling in callbacks is small let's just merge these together.

**Comment by MSeeker1340** [comment]
For some reason the first test (which passes on my local machine) fails on CI, so I'm just gonna flag it anyways. Someone should take a look at the multivariate geometric tests in the future.

Also updated the CI scripts to use Julia v1.0.

**Comment by ChrisRackauckas** [comment]
Multivariate geometric is using the average over 50 runs to calculate convergence. Try 100 or 200.

**Comment by MSeeker1340** [comment]
I added `Random.seed!` to ensure the first test passes. However the second test is still broken even after increasing to 100 runs (200 runs takes too long to run).

File: src/callbacks.jl [hunk 1]
```diff
@@ -196,8 +196,12 @@ function apply_callback!(integrator,callback::ContinuousCallback,cb_time,prev_si
     change_t_via_interpolation!(integrator,integrator.tprev+cb_time)
   end
 
+  # handle saveat
+  savevalues!(integrator)
```
Comment (ChrisRackauckas) [inline src/callbacks.jl]: Just for continuous callbacks?

Comment (MSeeker1340) [reply src/callbacks.jl]: Ah yes I forgot to include Yingbo's fix on discrete callbacks. Let me fix that.

**Comment by ChrisRackauckas** [comment]
It looks like the CI update is pointing out that there are some floating point issues coming up, but it's unrelated to this PR so I'll merge and handle it later.

**Pull request merged**